### ISSUE TITLE
Want to change dice coefficient function to class-balanced cross entropy function.

### ISSUE THREAD

**Issue by GodOfSmallThings** [argman/EAST#9]
I tried to change this code to class-balanced cross entropy function.

def dice_coefficient(y_true_cls, y_pred_cls,
                     training_mask):
    '''
    dice loss
    :param y_true_cls:
    :param y_pred_cls:
    :param training_mask:
    :return:
    '''
    eps = 1e-5
    intersection = tf.reduce_sum(y_true_cls * y_pred_cls * training_mask)
    union = tf.reduce_sum(y_true_cls * training_mask) + tf.reduce_sum(y_pred_cls * training_mask) + eps
    loss = 1. - (2 * intersection / union)
    tf.summary.scalar('classification_dice_loss', loss)
    return loss

However, I don't understand why there is training mask and what its' role is. I would be thankful if somebody tells :) Thanks

**Comment by argman** [comment]
the training mask is used to ignore some hard areas(like some blurred areas you can hardly tell whether its text or not)

**Comment by eragonruan** [comment]
the training mask is used to mask out some dont care area, such as small text. If you wanna use class balanced cross entropy, you can set y_true_cls and y_pred_cls of these dont care area to zero,which will not affect the cross entropy loss.

**Comment by GodOfSmallThings** [comment]
I got it. Thank you so much !
